src/roiextractors/extractors/miniscopeimagingextractor/miniscopeimagingextractor.py

```diff
@@ -0,0 +1,116 @@
+        miniscope_config_files = natsort.natsorted(
+            list(self.folder_path.glob(f"*/Miniscope/{configuration_file_name}"))
+        )
+        assert (
+            miniscope_config_files
+        ), f"The configuration files ({configuration_file_name} files) are missing from '{self.folder_path}'."
+
+        # Set the sampling frequency from the configuration file
+        with open(miniscope_config_files[0], newline="") as f:
+            self._miniscope_config = json.loads(f.read())
+        self._sampling_frequency = float(re.search(r"\d+", self._miniscope_config["frameRate"]).group())
+
+        imaging_extractors = []
+        for file_path in miniscope_avi_file_paths:
+            extractor = _MiniscopeImagingExtractor(file_path=file_path)
+            extractor._sampling_frequency = self._sampling_frequency
+            imaging_extractors.append(extractor)
+
+        super().__init__(imaging_extractors=imaging_extractors)
+
+
+class _MiniscopeImagingExtractor(ImagingExtractor):
+    extractor_name = "_MiniscopeImaging"
+
+    def __init__(self, file_path: PathType):
+        from neuroconv.datainterfaces.behavior.video.video_utils import VideoCaptureContext
+
+        self._video_capture = VideoCaptureContext
+        self._cv2 = get_package(package_name="cv2", installation_instructions="pip install opencv-python-headless")
+        self.file_path = file_path
+        super().__init__()
+
+        with self._video_capture(file_path=str(file_path)) as video_obj:
+            self._num_frames = video_obj.get_video_frame_count()
+            self._image_size = video_obj.get_frame_shape()
+            self._dtype = video_obj.get_video_frame_dtype()
+
+        self._sampling_frequency = None
+
+    def get_num_frames(self) -> int:
+        return self._num_frames
+
+    def get_num_channels(self) -> int:
+        return 1
+
+    def get_image_size(self) -> Tuple[int, int]:
+        return self._image_size[:-1]
+
+    def get_sampling_frequency(self):
+        return self._sampling_frequency
+
+    def get_dtype(self) -> DtypeType:
+        return self._dtype
+
+    def get_channel_names(self) -> List[str]:
+        return ["OpticalChannel"]
+
+    def get_video(
+        self, start_frame: Optional[int] = None, end_frame: Optional[int] = None, channel: int = 0
+    ) -> np.ndarray:
+        """
+        The grayscale conversion is based on minian
+        https://github.com/denisecailab/minian/blob/f64c456ca027200e19cf40a80f0596106918fd09/minian/utilities.py#LL272C12-L272C12
+        """
+        if channel != 0:
+            raise NotImplementedError(
+                f"The {self.extractor_name}Extractor does not currently support multiple color channels."
+            )
+
+        end_frame = end_frame or self.get_num_frames()
+        start_frame = start_frame or 0
+
+        video = np.empty(shape=(end_frame - start_frame, *self.get_image_size()))
```

Comment:
@weiglszonja I think you might need to set `dtype="int16"` here; the file shows the data ending up in `float64`

I'd also ensure that the type returned by `cv2.cvtColor`  with `cv2.COLOR_RGB2GRAY` is `int16` not `float64` (and manually cast if so)

Reply:
`float64` is definitely incorrect, thank you for catching it! 

> I'd also ensure that the type returned by cv2.cvtColor with cv2.COLOR_RGB2GRAY is int16 not float64 (and manually cast if so)

The return type is actually `uint8` and not int16 which is the same as the `VideoCaptureContext.get_video_frame_dtype()` returns

Reply:
From meeting: yes, I suppose it does. OK then